src/contour/Config.h

```diff
@@ -145,9 +145,23 @@ struct SimpleColorConfig
 
 using ColorConfig = std::variant<SimpleColorConfig, DualColorConfig>;
 
+// Represents a single SSH login configuration.
+struct SshLoginConfig
+{
+    std::string host;
+    uint16_t port = 22;
+    std::string user;
+    std::filesystem::path privateKeyFile;
+    std::filesystem::path publicKeyFile;
+    std::filesystem::path knownHostsFile;
+    bool forwardAgent = false;
+};
```

Comment:
- knownHostsFile gets defaulted in Config.cpp during config load time, and set to `~/.ssh/known_hosts` (with `~` being resolved to user's home dir).
- public key is not needed usually, and at least not on OpenSSL based setups (not sure what libssh2 is using on Windows, but there I was authenticating with ssh agent as well as priv-key only successfully, it seems
- private key, that's tricky, because what if there's more than one in `~/.ssh`? We could now go the ssh-agent way and just iterate through all of them and take the first that works. but I think that may not be desirable. The user (IMHO) should explicitly state what key to use to log to remote servers. What we could do, MAYBE, is, if there's only one priv key in `~/.ssh/`, then we default to that. But what if these key names follow a non-standard naming? That would lead to an interesting complexity at lesat (IMHO).